sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/_connection.py

```diff
@@ -82,6 +82,9 @@ class Connection(object):  # pylint:disable=too-many-instance-attributes
      keys: `'proxy_hostname'` (str value) and `'proxy_port'` (int value). When using these settings,
      the transport_type would be AmqpOverWebSocket.
      Additionally the following keys may also be present: `'username', 'password'`.
+    :keyword float socket_timeout_interval: The time in seconds that the underlying socket in the transport should
```

Comment:
In _pyamqp/client.py, there's already a socket_timeout keyword that takes Union[float, bool] which is passed into `connection.listen(wait=self._socket_timeout)`. This blocks on that socket until the frame arrives for `self._socket_timeout` time. Currently, that `self._socket_timeout` is set to `False` in the AMQPClient constructor. So should this float value just be passed through into that `connection.listen()`?

Reply:
Sounds like we have some bigger design questions then. There definitely shouldn't be two things called `socket_timeout` that are doing different things.

Reply:
ah yep, that makes sense. It was raising an error before with websocket but will investigate/fix and update so client has a single `socket_timeout` attribute.

Reply:
updated!

Reply:
Sorry - I didn't mean to imply that I thought these two properties should be merged.
I don't know if that's the case - that would need some investigation. Currently they're doing to separate things, I'm not sure if sharing the same property between them is the right thing to do.
We cannot have two separate properties both named `socket_timeout`, so the "bigger design questions" would be:
- Should this be two properties at all? If not, and we merge them, what does it mean for behaviour if I make this value bigger and we're applying to longer timeout in two different places? Or shorter? 
- If it should be two properties, then one of them will need to be renamed to clarify the distinction.

Reply:
TLDR; So did some investigation + discussion with Kashif, and found the following things:
- The `socket_timeout` and `socket_timeout_interval` _should_ be the same value. - `block_with_timeout` is unnecessary/redundanct, so we've removed it.
- for sync, the socket timeout is passed in when creating the connection and acts as the read/write timeout for both websocket and ssl transport.
- since the connect timeout has a different purpose, I've added a separate config/kwarg back to SSLTransport. We can expose this at some point in the future if we need to, but it is set to 1 by default for now.
- sync ws doesn't seem to take a separate timeout for connecting
- for async, read/write timeouts on both ssl and websocket are controlled by asyncio.wait_for(), which is where we pass in the socket_timeout.
- For async ssl, there is no separate connect_timeout.
- For async ws, we pass in a connect_timeout when creating the connection. If we don't, it's 10 secs by default.

More info on block_with_timeout:
In sync, this socket_timeout value is passed into connection.listen(...) all the way through to [`_read_frame`](https://github.com/Azure/azure-sdk-for-python/blob/09035248a8e9fa27ea07cd4cbcdde2f1b6efdcfc/sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/_connection.py#L246) and [`_send_frame`](https://github.com/Azure/azure-sdk-for-python/blob/09035248a8e9fa27ea07cd4cbcdde2f1b6efdcfc/sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/_connection.py#L273). Inside those methods, if a timeout value _has_ been passed in, then it calls the `[block_with_timeout](https://github.com/Azure/azure-sdk-for-python/blob/09035248a8e9fa27ea07cd4cbcdde2f1b6efdcfc/sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_transport.py#L205)` method on the transport. However, we found that this method is not used/useful with the way that we setup our socket. The way that we currently do this by just calling send/receive_frame, it [_already_ blocks with timeout b/c we call `sock.settimeout()` (which is a blocking call)](https://github.com/Azure/azure-sdk-for-python/blob/09035248a8e9fa27ea07cd4cbcdde2f1b6efdcfc/sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_transport.py#L348) when we _init_socket. That method was [used before when we used OS timeouts, rather than socket timeouts](https://github.com/Azure/azure-sdk-for-python/blob/09035248a8e9fa27ea07cd4cbcdde2f1b6efdcfc/sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_transport.py#L335), which I believe were non-blocking by default.